src/test/java/core/basesyntax/FruitServiceImplTests.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.impl.FruitServiceImpl;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.impl.BalanceOperation;
+import core.basesyntax.strategy.impl.OperationHandler;
+import core.basesyntax.strategy.impl.PurchaseOperation;
+import core.basesyntax.strategy.impl.ReturnOperation;
+import core.basesyntax.strategy.impl.SupplyOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitServiceImplTests {
+    private static FruitService fruitService;
+    private static List<FruitTransaction> fruitTransactions;
+
+    @BeforeClass
+    public static void beforeClass() {
+        FruitDao fruitDao = new FruitDaoImpl();
+        Map<FruitTransaction.Operation, OperationHandler> activityHandlerMap = new HashMap<>();
+        activityHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperation(fruitDao));
+        activityHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation(fruitDao));
+        activityHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperation(fruitDao));
+        activityHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperation(fruitDao));
+        OperationStrategy operationStrategy = new OperationStrategyImpl(activityHandlerMap);
+        fruitService = new FruitServiceImpl(operationStrategy);
+    }
+
+    @Before
+    public void setUp() {
+        fruitTransactions = new ArrayList<>();
+    }
+
+    @Test
+    public void processData_emptyFruitTransaction_ok() {
+        fruitService.processData(fruitTransactions);
+        Map<Fruit, Integer> expected = new HashMap<>();
+        Map<Fruit, Integer> actual = Storage.fruits;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void processData_oneFruitTransactionSupply_ok() {
```

Comment:
`oneFruitTransactionSupply` - why `one`?
did you mean per fruit?

Reply:
Yeah, bad naming, I changed it